src/Payload.php

```diff
@@ -60,6 +60,41 @@ public function getClaims()
     }
 
     /**
+     * Checks if a payload matches some expected values.
+     *
+     * @param  array  $values
+     * @param  bool  $strict
+     *
+     * @return bool
+     */
+    public function matches(array $values, $strict = false)
+    {
+        if (empty($values)) {
+            return false;
+        }
+
+        foreach ($values as $key => $value) {
+            if (($strict && $this->get($key) !== $value) || $this->get($key) != $value) {
```

Comment:
@tymondesigns I've changed things a bit - nevertheless when I see #746 (on which I'll have a deeper look), I feel like the claim itself should be responsible for this strict/not-strict match on itself

Let me know you thoughts on that !